src/pages/admin/leeszalenbeheer/leeszalen/index.tsx

```diff
@@ -0,0 +1,227 @@
+import { Column, Table, TableOptions } from '@meemoo/react-components';
+import { GetServerSideProps } from 'next';
+import { useTranslation } from 'next-i18next';
+import Head from 'next/head';
+import React, { FC, useCallback, useMemo } from 'react';
+import { useQueryParams } from 'use-query-params';
+
+import {
+	ADMIN_READING_ROOMS_OVERVIEW_QUERY_PARAM_CONFIG,
+	ReadingRoomsOverviewTableColumns,
+	ReadingRoomsOverviewTablePageSize,
+} from '@admin/const';
+import { AdminLayout } from '@admin/layouts';
+import { withAuth } from '@auth/wrappers/with-auth';
+import { withI18n } from '@i18n/wrappers';
+import { useGetReadingRooms } from '@reading-room/hooks/get-reading-rooms';
+import { VistorSpaceService } from '@reading-room/services';
+import { ReadingRoomOrderProps, ReadingRoomStatus } from '@reading-room/types';
+import { Loading, PaginationBar, SearchBar, sortingIcons } from '@shared/components';
+import { SEARCH_QUERY_KEY } from '@shared/const';
+import { toastService } from '@shared/services/toast-service';
+import { OrderDirection } from '@shared/types';
+import { createPageTitle } from '@shared/utils';
+
+const ReadingRoomsOverview: FC = () => {
+	const { t } = useTranslation();
+	const [filters, setFilters] = useQueryParams(ADMIN_READING_ROOMS_OVERVIEW_QUERY_PARAM_CONFIG);
+
+	const {
+		data: readingRooms,
+		isLoading,
+		refetch,
+	} = useGetReadingRooms(
+		filters.search,
+		filters.page,
+		ReadingRoomsOverviewTablePageSize,
+		filters.orderProp as ReadingRoomOrderProps,
+		filters.orderDirection as OrderDirection
+	);
+
+	// Filters
+
+	const sortFilters = useMemo(() => {
+		return [
+			{
+				id: filters.orderProp,
+				desc: filters.orderDirection !== OrderDirection.asc,
+			},
+		];
+	}, [filters]);
+
+	// Events
+
+	const onSortChange = useCallback(
+		(rules) => {
+			setFilters({
+				...filters,
+				orderProp: rules[0]?.id || undefined,
+				orderDirection: rules[0]
+					? rules[0].desc
+						? OrderDirection.desc
+						: OrderDirection.asc
+					: undefined,
+				page: 1,
+			});
+		},
+		[filters, setFilters]
+	);
+
+	// Callbacks
+	const onFailedRequest = () => {
+		refetch();
+
+		toastService.notify({
+			maxLines: 3,
+			title: t('pages/admin/leeszalenbeheer/leeszalen/index___er-ging-iets-mis'),
+			description: t(
+				'pages/admin/leeszalenbeheer/leeszalen/index___er-is-een-fout-opgetreden-tijdens-het-aanpassen-van-de-status'
+			),
+		});
+	};
+
+	const updateRoomStatus = (roomId: string, status: ReadingRoomStatus) => {
+		VistorSpaceService.update(roomId, {
+			status: status,
+		})
+			.catch(onFailedRequest)
+			.then((response) => {
+				if (response === undefined) {
```

Comment:
Might be safer to also check for null:
```suggestion
				if (!response) {
```